pyannote/audio/applications/change_detection.py

```diff
@@ -0,0 +1,420 @@
+                 mlp: [16]                   # LSTM is bidirectional
+                 bidirectional: 'concat'
+                 final_activation: 'sigmoid'
+
+        sequences:
+             duration: 3.2                 # this experiments relies
+             step: 0.8                     # on sliding windows of 3.2s
+             balance: 0.05                 # with a step of 0.8s
+        ...................................................................
+
+"train" mode:
+        First, one should train the raw sequence labeling neural network using
+        "train" mode. This will create the following directory that contains
+        the pre-trained neural network weights after each epoch:
+                <experiment_dir>/train/<database.task.protocol>.<subset>
+
+        This means that the network was trained on the <subset> subset of the
+        <database.task.protocol> protocol. By default, <subset> is "train".
+        This directory is called <train_dir> in the subsequent "evaluate" and "apply" mode.
+
+"evaluate" mode:
+        Then, one can evaluate the model with "evaluate" mode. This will create 
+        the following directory that contains coverages and purities based 
+        on different thresholds:
+                <train_dir>/evaluate/<database.task.protocol>.<subset>
+        This means that the model is evaluated on the <subset> subset of the
+        <database.task.protocol> protocol. By default, <subset> is "development".
+
+"apply" mode
+        Finally, one can apply speaker change detection using "apply" mode.
+        This will create the following files that contains the segmentation results 
+        with a predetermined threshold:
+                <tune_dir>/segments/<database.task.protocol>.<subset>/<threshold>/{uri}.0.seg
+        This means that file whose unique resource identifier is {uri} has been
+        processed.
+
+"""
+
+import yaml
+import pickle
+import os.path
+import functools
+import numpy as np
+
+from docopt import docopt
+
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+
+import pyannote.core
+import pyannote.core.json
+
+from pyannote.audio.labeling.base import SequenceLabeling
+from pyannote.audio.generators.change import ChangeDetectionBatchGenerator
+
+from pyannote.audio.labeling.aggregation import SequenceLabelingAggregation
+from pyannote.audio.signal import Peak
+
+from pyannote.database import get_database
+from pyannote.audio.optimizers import SSMORMS3
+
+from pyannote.metrics.segmentation import SegmentationPurity
+from pyannote.metrics.segmentation import SegmentationCoverage
+from pyannote.metrics import f_measure
+
+from pyannote.database.util import FileFinder
+from pyannote.database.util import get_unique_identifier
+from pyannote.database import get_protocol
+
+from pyannote.audio.util import mkdir_p
+
+
+
+def train(protocol, experiment_dir, train_dir, subset='train'):
+
+    # -- TRAINING --
+    batch_size = 1024
+    nb_epoch = 2
```

Comment:
After this last round of changes, I think we are ready for the merge: can you switch to 100 epochs?